os/arch/arm/src/imxrt/imxrt_mpuinit.c

```diff
@@ -97,8 +97,8 @@ const struct mpu_region_info regions_info[] = {
 /****************************************************************************
  * Public Variables
  ****************************************************************************/
-#ifdef CONFIG_APP_BINARY_SEPARATION
-uint32_t g_app_mpu_region;
+#if defined(CONFIG_BUILD_PROTECTED) || defined(CONFIG_APP_BINARY_SEPARATION)
+uint32_t g_mpu_region_nr = 0;
```

Comment:
how about getting the region number through return value of mpu function?
When someone uses app binary separation, he should know what g_mpu_region_nr is. But it is impossible to analyze of our mpu code. I suggest that assignment as nuttx (automatically), we could get the some specific number through return value.

Reply:
I agree that we could do it as you mentioned. However, let me make those changes as a new PR. This PR already has been modified many times. So, could you please merge this PR and i will raise a new PR for MPU.

Reply:
ok